scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -1591,7 +1591,7 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
 
   def unquoteXmlExpr(): Term = {
     // TODO: verify this
-    dropAnyBraces(expr())
+    unquoteExpr()
   }
```

Comment:
If I remove `dropAnyBraces` in `unquoteXmlPattern` then it fails to parse xml literals in pattern position.
Also I added a test to make sure that empty splices in pattern position fail to parse (like in scalac):
```scala
checkError("e match { case <a>{}</a> => ??? }")
```